apps/voting/src/shared/store.ts

```diff
@@ -119,48 +122,115 @@ const electionConfigBaseAtom = atom<ElectionConfig | undefined>(undefined);
+        const config: ElectionConfig = {
+            ...preliminaryConfig,
             candidates: candidates.filter(isDefined),
-            description: config.election_description,
-            manifest,
-            parameters,
-            guardianKeys: config.guardian_keys,
-            setupDone: false,
+            manifest: manifest,
+            parameters: parameters,
         };
+        set(electionConfigBaseAtom, config);
+    },
+);
 
-        set(electionConfigBaseAtom, mappedConfig);
+/**
+ * Primitive atom for holding the {@linkcode ElectionConfig} of the election contract
+ */
+const guardiansStateBaseAtom = atom<GuardiansState | undefined>(undefined);
 
-        const guardians = await getGuardiansState();
-        if (guardians === undefined) {
-            throw new Error('Failed to get election config');
+/**
+ * Holds the configuration of the election contract. A reference to this should always be kept in the application root
+ * to avoid having to fetch the configuration more than once.
+ */
+export const guardiansStateAtom = atom(
+    (get) => get(guardiansStateBaseAtom),
+    async (_, set) => {
+        const hydrate = CONFIG.contractConfig !== undefined;
+        let config: GuardiansState;
+        if (hydrate) {
+            config = {
+                guardianKeys: CONFIG.contractConfig!.guardian_keys,
+                setupDone: CONFIG.contractConfig!.guardians_setup_done,
+            };
+        } else {
+            const guardians = await getGuardiansState();
+            if (guardians === undefined) {
+                throw new Error('Failed to get election config');
+            }
+
+            const setupDone = guardians.every(
+                ([, g]) =>
+                    g.public_key.type === 'Some' &&
+                    g.encrypted_share.type === 'Some' &&
+                    g.status.type === 'Some' &&
+                    g.status.content.type === 'VerificationSuccessful',
+            );
+            const guardianKeys = guardians
+                .map(([, g]) => {
+                    if (g.public_key.type === 'None') return undefined;
+                    return g.public_key.content;
+                })
+                .filter(isDefined);
+            config = {
+                guardianKeys,
+                setupDone,
+            };
         }
 
-        const setupDone = guardians.every(
-            ([, g]) =>
-                g.public_key.type === 'Some' &&
-                g.encrypted_share.type === 'Some' &&
-                g.status.type === 'Some' &&
-                g.status.content.type === 'VerificationSuccessful',
-        );
+        set(guardiansStateBaseAtom, config);
+    },
+);
 
-        set(electionConfigBaseAtom, { ...mappedConfig, setupDone });
+const electionResultBaseAtom = atom<ElectionResultResponse>(undefined);
+
+/**
+ * Holds the election result from the election contract, if available. Invoke setter to refresh the value from the
+ * contract
+ */
```

Comment:
I attempt to rephrase this (not sure if my below wordings are more confusing).

```suggestion
/**
 * Holds the election result from the election contract, if available. Otherwise, invokes the setter in an attempt to fetch/refresh the election results from the election contract.
 */
```

Reply:
So the way this is meant to be understood is: 
- If you invoke the atom getter, the election config is returned
- If you invoke the atom setter, the election config is updated from the contract